feat: add organization details to audit log response

code-asher opened the pull request:
This is so the frontend will be able to display the organization name in the audit log row.

I added `MinimalOrganization` to mirror `MinimalUser`.  It will be omitted if there was no organization, which I felt made more sense in the frontend (rather than the uuid zero value), but I can change that back if we want.

This new object replaces the old organization ID property on audit logs.  Technically a breaking change?  But we were not using it and I doubt anyone else was.

This is explained in the commit message, but `params.OrganizationID == uuid.Nil` was removed from the test audit log creation so I could test audit logs without an org.

Part 1 of https://github.com/coder/coder/issues/13913

code-asher committed:
Allow creating test audits with nil org

Not all audit entries have organization IDs, so this will allow us to
test those cases.

code-asher committed:
Add organization details to audit log queries

code-asher committed:
Add organization to audit log response

This replaces the old ID.  This is a breaking change but organizations
were not being used before.

code-asher commented on coderd/database/queries/auditlogs.sql:
```diff
@@ -18,6 +18,9 @@ SELECT
     users.deleted AS user_deleted,
     users.theme_preference AS user_theme_preference,
     users.quiet_hours_schedule AS user_quiet_hours_schedule,
+    COALESCE(organizations.name, '') AS organization_name,
+    COALESCE(organizations.display_name, '') AS organization_display_name,
+    COALESCE(organizations.icon, '') AS organization_icon,
```
Could maybe be NULL instead, not sure it matters that much though?

code-asher replied:
Awesome, good to know thanks!

code-asher commented on codersdk/organizations.go:
```diff
@@ -39,18 +39,22 @@ func ProvisionerTypeValid[T ProvisionerType | string](pt T) error {
 
+// Organization is the JSON representation of a Coder organization.
+type Organization struct {
+	MinimalOrganization `table:"m,recursive_inline"`
```
Copied from `MinimalUser`,  I see that `recursive_inline` skips prefexing the parent name but I am not sure what the `m` means here.

code-asher replied:
Ahhhh gotcha, thanks!

Emyrk commented on codersdk/audit.go:
```diff
@@ -125,14 +125,13 @@ type AuditDiffField struct {
-	UserAgent      string       `json:"user_agent"`
-	ResourceType   ResourceType `json:"resource_type"`
-	ResourceID     uuid.UUID    `json:"resource_id" format:"uuid"`
+	ID           uuid.UUID    `json:"id" format:"uuid"`
+	RequestID    uuid.UUID    `json:"request_id" format:"uuid"`
+	Time         time.Time    `json:"time" format:"date-time"`
+	IP           netip.Addr   `json:"ip"`
+	UserAgent    string       `json:"user_agent"`
+	ResourceType ResourceType `json:"resource_type"`
+	ResourceID   uuid.UUID    `json:"resource_id" format:"uuid"`
```
To prevent this PR from being a breaking change, can we add back the `OrganizationID` field too? I know it is redundant, but in case anyone is using this field at the moment.

I have another issue I need to break the api type: https://github.com/coder/coder/issues/13949

Might drop this field and with the others in 1 release, rather than 2. 


```suggestion
	ID           uuid.UUID    `json:"id" format:"uuid"`
	RequestID    uuid.UUID    `json:"request_id" format:"uuid"`
	Time         time.Time    `json:"time" format:"date-time"`
	IP           netip.Addr   `json:"ip"`
	UserAgent    string       `json:"user_agent"`
	ResourceType ResourceType `json:"resource_type"`
	ResourceID   uuid.UUID    `json:"resource_id" format:"uuid"`
	// OrganizationID
	// Deprecated: Use 'organization.id' instead
	OrganizationID uuid.UUID    `json:"organization_id" format:"uuid"`
```

code-asher replied:
Yup, will do.

code-asher commented on coderd/audit_test.go:
```diff
@@ -31,7 +31,8 @@ func TestAuditLogs(t *testing.T) {
 
 		err := client.CreateTestAuditLog(ctx, codersdk.CreateTestAuditLogRequest{
-			ResourceID: user.UserID,
+			OrganizationID: uuid.New(),
```
Oh interesting.  I did not really think too hard on this since I was just moving out the `uuid.New()` from `CreateTestAuditLog()` out to the test, but you make a good point that most of these do not even need to set the ID  for the test to pass.  I will remove them!

code-asher committed:
Clarify user/organization audit log tests

There is another test called "organizations" so this distinguishes that
this is checking the response itself contains this field.  Rename the
user test to match.

code-asher committed:
Remove unnecessary org ID from audit tests

code-asher committed:
Add back OrganizationID to audit logs

But mark it as deprecated.
